clients/src/main/java/org/apache/kafka/common/record/RecordBatch.java

```diff
@@ -58,7 +58,7 @@
      * Used to indicate an unknown leader epoch, which will be the case when the record set is
      * first created by the producer.
      */
-    int UNKNOWN_PARTITION_LEADER_EPOCH = -1;
+    int UNKNOWN_PARTITION_LEADER_EPOCH = -2;
```

Comment:
Out of curiosity, why do we use `-2` instead of `-1` here? It may be worth adding a comment explaining.